### PR TITLE
chore: add .env.example for easier environment setup and refine .gitignore

### PR DESCRIPTION
**Pull request by piotrski**
I’ve made a couple of updates here to developer experience:
 - Added `.env.example` for easier environment setup
This provides a template for environment variables, helping developers get their local setup configured quickly.

- Updated `.gitignore` to exclude Azurite local storage files
We’ve added patterns to ignore Azurite's local storage files `(__azurite_db_*, __blobstorage__, __queuestorage__)`, keeping the working directory cleaner during local development.

### CODE REVIEW

**Comment by piotrski** [comment]
@microsoft-github-policy-service agree company="Callstack"

File: api/.env.example [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+DISABLE_MANAGEMENT=false
+# Set to 'false' in order to disable account registration
+ENABLE_ACCOUNT_REGISTRATION=true
+# Set to the max number of megabytes allowed for file uploads
+UPLOAD_SIZE_LIMIT_MB=200
+
+# To enable generating diffs for releases
+ENABLE_PACKAGE_DIFFING=false
+
+# To enable KeyVault credential resolution
```
Comment (lucen-ms) [inline api/.env.example:69]: Something is missing here.

File: api/README.md [hunk 2]
```diff
@@ -16,11 +16,26 @@ Additionally, you need to specify [EMULATED](ENVIRONMENT.md#emulated) flag equal
...
-1. Build the server by running `npm run build`.
-1. Start the server by running `npm run start:env`.
+
+1. Clone the CodePush Service repository to your local machine.
+
+2. Copy the `.env.example` file to a new file named `.env` in the root directory:
+   ````bash
+   cp .env.example .env
+   ````
+   Fill in the values for each environment variable in the `.env` file according to your development or production setup.
```
Comment (lucen-ms) [inline api/README.md:26]: Only mandatory variables are needed - please mark mandatory values in `.env.example` file.

**Review by lucen-ms — CHANGES_REQUESTED** [review]

File: api/README.md (hunk 2 above)
Comment (sanduluca) [reply api/README.md:26]: It’s best to list all environment variables that the app uses, even if some are optional. This ensures transparency and clarity for developers who may work on or deploy the application in different environments.

How exactly should be "marked" the mandatory values ?

Comment (lucen-ms) [reply api/README.md:26]: Up to you - e.g.

```
# Required, the URL of your server
SERVER_URL=http://localhost:3000
```

**Comment by lucen-ms** [comment]
@piotrski will you be finishing the PR please?

**Comment by piotrski** [comment]
@lucen-ms apologies, it must have slipped my mind. I'll take a look at this later today.

**Comment by piotrski** [comment]
I've updated the formatting of the .env.example file to make it cleaner and more organized. Additionally, I've added information indicating which variables are required, as you requested.

Let me know if you have any feedback or if further adjustments are needed!

File: api/.env.example [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+REDIS_HOST=     # Redis server IP address
+REDIS_PORT=6379 # Redis port (default: 6379)
+REDIS_KEY=      # Redis authentication key
+
+# ==============================
+# Unit Testing Configuration
+# ==============================
+TEST_AZURE_STORAGE=false  # Run API unit tests against Azure storage
+AZURE_STORAGE_ACCOUNT=    # Storage account for testing (if TEST_AZURE_STORAGE=true)
+AZURE_STORAGE_ACCESS_KEY= # Storage access key for testing
```
Comment (lucen-ms) [inline api/.env.example:55]: AZURE_STORAGE_ACCOUNT=    # Storage account for testing (if TEST_AZURE_STORAGE=true)
    AZURE_STORAGE_ACCESS_KEY= # Storage access key for testing

This is duplication of the same settings above which will override them. Every setting should be only once in the example.

Comment (piotrski) [reply api/.env.example]: You're right, I just removed them.

**Review by lucen-ms — APPROVED** [review]

**Comment by lucen-ms** [comment]
Thank you @piotrski for the contribution - I've merged your changes.

**Pull request merged**